cmd/http/main.go

```diff
@@ -61,7 +66,21 @@ func main() {
 	}
 	defer nc.Close()
 
-	conn, err := grpc.Dial(cfg.ThingsURL, grpc.WithInsecure())
+	secureOption := grpc.WithInsecure()
+	if cfg.CACerts != "" {
+		thingsURL, err  := url.Parse(cfg.ThingsURL)
+		if err != nil {
+			logger.Error(fmt.Sprintf("Failed to parse ThingsURL: %s", err))
+		}
+		tpc, err := credentials.NewClientTLSFromFile(cfg.CACerts, thingsURL.Host)
```

Comment:
If `url.Parse(cfg.ThingsURL)` fails, `thingsURL` is possible an empty value, or nil, in case of the pointer, so calling `thingsURL.Host` is dangerous.